rotkehlchen/globaldb/manual_price_oracles.py

```diff
@@ -55,26 +60,41 @@ class ManualCurrentOracle(CurrentPriceOracleInterface):
 
     def __init__(self) -> None:
         super().__init__(oracle_name='manual current price oracle')
+        self.database: Optional['DBHandler'] = None
+
+    def set_database(self, database: 'DBHandler') -> None:
+        self.database = database
 
     def rate_limited_in_last(self, seconds: Optional[int] = None) -> bool:
         return False
 
-    def query_current_price(self, from_asset: Asset, to_asset: Asset) -> Price:
+    def query_current_price(  # type: ignore  # due to incompatibility with the supertype
+            self,
+            from_asset: Asset,
+            to_asset: Asset,
+            match_main_currency: bool,
```

Comment:
What is this argument for? You don't explain it in the docstring and I don't really get it from the code.

When is it `True` in production code? Shouldn't this always be true for this fix/hack to work?

From what I understand this argument should not exist

Reply:
This should be `True` only when frontend calls the price to display it to the user. But (if I am not mistaken), we should not take this "hack" into account when, for example, saving a snapshot (since we need value in usd).

Reply:
We don't want that at snapshot saving for the moment so we indeed should have a way to differentiate

Then in that case you need to:

1. properly explain this in the docstring. You have to get used to explaining things in docstring.
2. Move the if condition before even opening the cursor and reading the DB. The logic you wrote will always open a cursor and read the main currency from the DB and ony then check if it should have done it. Which is the main reason I did not understand why you are doing this.